### ISSUE TITLE
Occasionally plex is reporting failures in post processing.

### ISSUE THREAD

**Issue by CollabSensei** [cedarrapidsboy/pms-postprocessing#5]
```May 16, 2020 13:00:21.045 [0x7f010395f700] DEBUG - Jobs: '/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Scripts/transcode_internal.sh' exit code for process 22310 is 0 (success)
May 16, 2020 13:00:21.045 [0x7f00a2ffd700] WARN - SLOW QUERY: It took 13310.000000 ms to retrieve 0 items.
May 16, 2020 13:00:21.051 [0x7f00717fa700] DEBUG - buildLiveM3U8: min 0 max 19 ended 0
May 16, 2020 13:00:21.052 [0x7f00eaffd700] DEBUG - Request: [127.0.0.1:48410 (Loopback)] GET /livetv/sessions/764106e7-73c7-422a-986b-924709a3e87c/0b8ffcad-2ee6-488f-af91-41d24325a350/00018.ts (103 live) Signed-in
May 16, 2020 13:00:21.052 [0x7f00eaffd700] DEBUG - Content-Length of /media/PlexMedia1/transcode/Transcode/Sessions/plex-transcode-764106e7-73c7-422a-986b-924709a3e87c/media-00018.ts is 1754228 (of total: 1754228).
May 16, 2020 13:00:21.052 [0x7f00697fa700] DEBUG - Request: [127.0.0.1:48286 (Loopback)] GET /livetv/sessions/764106e7-73c7-422a-986b-924709a3e87c/0b8ffcad-2ee6-488f-af91-41d24325a350/00019.ts (103 live) Signed-in
May 16, 2020 13:00:21.052 [0x7f00697fa700] DEBUG - Content-Length of /media/PlexMedia1/transcode/Transcode/Sessions/plex-transcode-764106e7-73c7-422a-986b-924709a3e87c/media-00019.ts is 1766636 (of total: 1766636).
May 16, 2020 13:00:21.054 [0x7f00f29f5700] DEBUG - Completed: [127.0.0.1:48410] 200 GET /livetv/sessions/764106e7-73c7-422a-986b-924709a3e87c/0b8ffcad-2ee6-488f-af91-41d24325a350/00018.ts (103 live) 2ms 1754228 bytes (pipelined: 17)
May 16, 2020 13:00:21.054 [0x7f00f29f5700] DEBUG - Removed transcode data consumer, active count 2 => 1
May 16, 2020 13:00:21.055 [0x7f00f31f6700] DEBUG - Completed: [127.0.0.1:48286] 200 GET /livetv/sessions/764106e7-73c7-422a-986b-924709a3e87c/0b8ffcad-2ee6-488f-af91-41d24325a350/00019.ts (103 live) 3ms 1766636 bytes (pipelined: 3)
May 16, 2020 13:00:21.055 [0x7f00f31f6700] DEBUG - Removed transcode data consumer, active count 1 => 0
May 16, 2020 13:00:21.057 [0x7f0062fed700] ERROR - Recorder: Exception caught looking for post-processed file: boost::filesystem::directory_iterator::construct: No such file or directory: "/media/PlexMedia1/TVShows/.grab/baf1fe599e3f8add2fcf8741bf6a0c8f0e5df679-c51305b7fd58fd3c0df24350139664739866cda3"
May 16, 2020 13:00:21.057 [0x7f0062fed700] ERROR - Recorder: Postprocessing script '/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Scripts/transcode_internal.sh' seems to have deleted the file. Silly script.
May 16, 2020 13:00:21.057 [0x7f0062fed700] DEBUG - Grabber: Scheduling 'Guy's Grocery Games - E6 - Redemption Tournament: Part 4 (4/5)' for assimilation.
May 16, 2020 13:00:21.058 [0x7f0062fed700] DEBUG - Subscription: Resolved subscription to library item 'Guy's Grocery Games' (22361)
May 16, 2020 13:00:21.059 [0x7f0062fed700] DEBUG - There were 1 top-level paths for Guy's Grocery Games.
May 16, 2020 13:00:21.059 [0x7f0062fed700] DEBUG - Assimilator: Postprocessing, we're going to put `Guy's Grocery Games - E6 - Redemption Tournament: Part 4 (4/5)` in `/media/PlexMedia1/TVShows/Guy's Grocery Games/Season 09/Guy's Grocery Games - S09E06 - Redemption Tournament Part 4 (4 5).ts`
May 16, 2020 13:00:21.059 [0x7f0062fed700] ERROR - Assimilator: Error processing files: boost::filesystem::directory_iterator::construct: No such file or directory: "/media/PlexMedia1/TVShows/.grab/baf1fe599e3f8add2fcf8741bf6a0c8f0e5df679-c51305b7fd58fd3c0df24350139664739866cda3"
May 16, 2020 13:00:21.059 [0x7f0062fed700] DEBUG - Grabber: Operation for 'Guy's Grocery Games - E6 - Redemption Tournament: Part 4 (4/5)' on channel 6102 completed with status error (An error occurred while postprocessing the recording)
May 16, 2020 13:00:21.070 [0x7f0062fed700] DEBUG - Activity: Ended activity 024025dc-eda9-49bf-9754-23399ce045c4.```
---
Attached Files
[Plex Media Server.log](https://github.com/cedarrapidsboy/pms-postprocessing/files/4639314/Plex.Media.Server.log)
[plex -journalctl.txt](https://github.com/cedarrapidsboy/pms-postprocessing/files/4639315/plex.-journalctl.txt)

**Comment by cedarrapidsboy** [comment]
Thanks for the report. This doesn't happen at all for me (at least, not yet). All my records are .TS streams. Not sure why that matters. I'll take another look at the program flow and see if I can find any more insight into the post processing inputs and outputs.

**Comment by cedarrapidsboy** [comment]
Did a bit of a refactor to reduce the working file remuxing. See the tag:

https://github.com/cedarrapidsboy/pms-postprocessing/releases/tag/v3-pre1

I don't know for sure why this is failing for you... I cannot reproduce it. I checked episodes with special characters in the name, shows with special characters. I have test cases where the original TS is deleted and replaced with an MP4, or MKV, and as long as I don't change the base filename, Plex doesn't complain.

I changed some variable quoting and some order of the operations. Perhaps something in this build will help.
